### DISCUSSION TITLE
Packages version compatiability

### DISCUSSION THREAD

**Discussion by ChettakattuA** [climateinteractive/SDEverywhere#323] [Q&A] [answered]
I tried installing SDEverywhere in windows and i wanna test it out but i keep running to this one error with pnpm test command and am unsure is it the packages with SDEverywhere or with pnpm... The error am getting is 

ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @sdeverywhere/build@0.3.1 test: `vitest run`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
ERROR: "test:pkgs" exited with 1.
 ELIFECYCLE  Test failed. See above for more details.

Can you help me out with this?
Thank you

**Comment by chrispcampbell** [comment]
Hi there, if you're looking to contribute to SDEverywhere or run the test models, it is necessary to do a one-time "build" of all packages.  In other words, be sure to do `pnpm build` before running `pnpm test`.

More details on this wiki page:
https://github.com/climateinteractive/SDEverywhere/wiki/Contributor-Guide#test-your-setup

If it's not working on your machine, please provide more details in this thread showing what commands you tried, full console output, etc.

Note that I'm in the middle of rewriting SDEverywhere documentation (and moving various sections from the README to the wiki).  If you want to see the latest in-development version of the README, you can find that on a branch here:
https://github.com/climateinteractive/SDEverywhere/blob/chris/265-readme/README.md

In particular, if you are looking to use SDEverywhere on your own SD model (or on the SIR or Hello World models in the `examples` directory), check out the Quick Start instructions:
https://github.com/climateinteractive/SDEverywhere/blob/chris/265-readme/README.md#quick-start

**Reply by ChettakattuA** [reply]
So i tried the pnpm build command with windows but still the build failed


> @sdeverywhere/hello-world@1.0.0 build C:\Users\ChettakattuA\Documents\SDE3\SDEverywhere\examples\hello-world
> sde bundle

node:internal/errors:490
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\ChettakattuA\Documents\SDE3\SDEverywhere\packages\cli\node_modules\@sdeverywhere\build\dist\index.js' imported from C:\Users\ChettakattuA\Documents\SDE3\SDEverywhere\packages\cli\src\sde-bundle.js
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:226:11)
    at moduleResolve (node:internal/modules/esm/resolve:839:10)
    at defaultResolve (node:internal/modules/esm/resolve:1069:11)
    at DefaultModuleLoader.resolve (node:internal/modules/esm/loader:305:12)
    at DefaultModuleLoader.getModuleJob (node:internal/modules/esm/loader:156:32)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:33)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.2.0
 ELIFECYCLE  Command failed with exit code 1.

**Reply by ChettakattuA** [reply]
Also i tried with virtual machine for ubuntu but still am facing some issues

chettakattua@chettakattua-VirtualBox:~/Documents/SDE-Test-1/SDEverywhere/examples/hello-world/model$ pnpm build

> project@1.0.0 build /home/chettakattua/Documents/SDE-Test-1/SDEverywhere/examples/hello-world/model
> sde bundle

Generating files...
Done generating files (0.1s)
Generating model...
  Generating WebAssembly module
Done generating model (2.4s)
Building worker
Copying changed files into place...
Done copying files
Generating model check bundle...
vite v3.2.7 building for production...
✓ 53 modules transformed.
../../../../../../../sde-prep/check-bundle.js   78.26 KiB / gzip: 27.61 KiB
Generating model check test configuration...
vite v3.2.7 building for production...
✓ 273 modules transformed.
../../../../../../../sde-prep/check-tests.js   212.75 KiB / gzip: 57.29 KiB
Running model checks...
ERROR: Cannot read properties of undefined (reading 'fromText')

 ELIFECYCLE  Command failed with exit code 1.

I believe you have raised the same issue #230 but the solution doesn't seems to work

**Reply by chrispcampbell** [reply, marked as answer]
Sorry for the difficulties you're encountering.  I'll see if I can reproduce the issues locally on Windows or Ubuntu.

In the meantime, I would recommend trying the examples outside of the `SDEverywhere` directory on your machine.  Working inside the `SDEverywhere` directory is only needed for contributors/developers of SDE, and since it's a monorepo, it's a little bit more involved to set up and diagnose issues.  If you instead use the SDEverywhere packages as an end user would (installing from the npm registry instead of using locally built ones) things are more likely to work.

To try that approach (ideally on Ubuntu, but Windows should work too), copy the `SDEverywhere/examples/hello-world` directory to some other place on your machine, for example the `SDE-Test-1` directory.  Then change into that directory and follow the Quick Start instructions I linked to above:

```sh
cd SDE-Test-1/hello-world
npm create @sdeverywhere
```

Let me know if this works for you.  That will help me to understand if the problems you faced earlier are related to the steps needed to build locally, or if there's a more general issue.  Thanks.

**Reply by ChettakattuA** [reply]
OKAY so Thank you for reply and Yes the example Hello-world worked outside of the SDEverywhere directory (I used Ubuntu, no sure about Windows). And it would be helpful for others too if you can mention this in documentation.

**Reply by chrispcampbell** [reply]
Thank you for testing and confirming.  The goal of the new documentation is to make it more focused on end users, so I'll make sure to clarify this in the new docs.
